### PR TITLE
Works okta

### PR DESCRIPTION
**Pull request by suqin-haha**
1. move filter out from internal
2. expose ReferenceContains Function
3. add debug code

### CODE REVIEW

**Comment by suqin-haha** [comment]
And revert the path prefix MR that merged before

File: filter/op_boolean_test.go [hunk 1]
```diff
@@ -2,15 +2,17 @@ package filter_test
 
 import (
 	"fmt"
-	internal "github.com/elimity-com/scim/internal/filter"
-	"github.com/elimity-com/scim/schema"
-	"github.com/scim2/filter-parser/v2"
 	"testing"
+
+	"github.com/elimity-com/scim/filter"
+
```
Comment (muir) [inline filter/op_boolean_test.go]: You don't need this blank line.

File: filter/op_binary.go [hunk 2]
```diff
@@ -2,7 +2,7 @@ package filter
 
 import (
 	"fmt"
-	"github.com/scim2/filter-parser/v2"
+	fp "github.com/scim2/filter-parser/v2"
```
Comment (muir) [inline filter/op_binary.go:5]: Separate stdlib includes from library includes

File: filter/op_boolean.go [hunk 3]
```diff
@@ -3,7 +3,7 @@ package filter
 import (
 	"fmt"
 	"github.com/elimity-com/scim/schema"
-	"github.com/scim2/filter-parser/v2"
+	fp "github.com/scim2/filter-parser/v2"
```
Comment (muir) [inline filter/op_boolean.go:6]: Ditto

File: filter/op_datetime.go [hunk 4]
```diff
@@ -3,7 +3,7 @@ package filter
 import (
 	"fmt"
 	datetime "github.com/di-wu/xsd-datetime"
-	"github.com/scim2/filter-parser/v2"
+	fp "github.com/scim2/filter-parser/v2"
```
Comment (muir) [inline filter/op_datetime.go:6]: Oh, I see. The original library mixes them. Okay, I don't care then.

**Review by muir — APPROVED** [review]
So this all seems fine, but it doesn't seem like a change you can easily feed upstream. You would need to describe it better to send it upstream.

**Pull request merged**